tests/integration/test_schema.py

```diff
@@ -1128,41 +1138,39 @@ async def test_schema(registry_async_client, trail):
     assert res.json()["error_code"] == 500
     assert res.json()["message"] == "Internal Server Error"
 
-    # nonexistent schema id
+
+async def test_schema_non_existing_id(registry_async_client):
+    """
+    Tests getting a non-existing schema id
+    """
     result = await registry_async_client.get(os.path.join("schemas/ids/123456789"))
     assert result.json()["error_code"] == 40403
 
-    # invalid schema_id
+
+@pytest.mark.parametrize("trail", ["", "/"])
+async def test_schema_non_invalid_id(registry_async_client, trail):
+    """
+    Tests getting an invalid schema id
+    """
     result = await registry_async_client.get(f"schemas/ids/invalid{trail}")
     assert result.status == 404
     assert result.json()["error_code"] == 404
     assert result.json()["message"] == "HTTP 404 Not Found"
 
-    res = await registry_async_client.post(
-        "subjects/{}/versions".format(subject), json={"schema": "{\"type\": \"string\", \"foo\": \"string\"}"}
-    )
-    assert res.status_code == 200
-    assert "id" in res.json()
-    assert schema_id != res.json()["id"]
-
-    # Fetch the schema back to see how it was mangled
-    result = await registry_async_client.get(os.path.join("schemas/ids/{}".format(res.json()["id"])))
-    schema = jsonlib.loads(result.json()["schema"])
-    assert schema["type"] == "string"
-    assert schema["foo"] == "string"
-
-    res = await registry_async_client.get("subjects")
-    assert res.status_code == 200
-    assert subject in res.json()
 
-    res = await registry_async_client.get("subjects/{}/versions".format(subject))
-    assert res.status_code == 200
-    assert res.json() == [1, 2]
+@pytest.mark.parametrize("trail", ["", "/"])
+async def test_schema_subject_invalid_id(registry_async_client, trail):
+    """
+    Creates a subject with a schema and trying to find the invalid versions for the subject.
+    """
+    subject = create_subject_name_factory(f"test_schema_subject_invalid_id-{trail}")()
+    unique_field_factory = create_field_name_factory(trail)
 
-    res = await registry_async_client.get("subjects/{}/versions/1".format(subject))
+    res = await registry_async_client.post(
+        "subjects/{}/versions".format(subject),
```

Comment:
```suggestion
        f"subjects/{subject}/versions"
```
nitpick: smaller expressions are usually easier to read with f-strings, it would also make the coder consistent to use mostly one style. I would use this on most of the places instead of the operator `%`

Reply:
This `"subjects/{}/versions".format(subject)` pattern is there already in master everywhere, it's not that this PR introduces them (at least too many). I'd use f-strings too, but is it worth refactoring (now)? That can be done later too.

Yes, it'd be best to use one string formatting mechanism. Unfortunately f-strings have an issue with curly braces. In order to have a literal curly brace, you need to have two of those. This makes strs containing JSON awkward to write using f-strings,e.g.

```
'{"type": "string", "unique": "%s"}' % unique_field_factory()
```

This would need to be something like this using f-strings:

```
f'{{"type": "string", "unique": "{unique_field_factory()}"}}'
```

So much stuff in the end of the line..... And it gets even worse if use double-quotes for the string.

```
f'{{"type": "string", "unique": "{unique_field_factory()}"}}'
```

The `%` operator is `format` method looks cleaner here.

Reply:
So, I'm not changing these right now.